src/webp_converter.js

```diff
@@ -0,0 +1,30 @@
+/* eslint-disable no-console */
+
+const fs = require('fs');
+const imagemin = require('imagemin');
+const imageminWebp = require('imagemin-webp');
+const path = require('path');
+
+
+// Recursively converts all jpg and png files to webp in docs directory tree
+(async () => {
+    async function convertImagesToWebp(dirPath) {
+        // Converts all jpg and png files in current directory to webp
+        await imagemin([`${dirPath}/*.{jpg,png}`], {
+            destination: dirPath,
+            plugins: [
+                imageminWebp({ quality: 65 }),
+            ],
+        });
+
+        const files = fs.readdirSync(dirPath);
+
+        // Find directories in current directory and convert their images to webp
+        await Promise.all(files.filter(file => fs.statSync(`${path.join(path.resolve(path.dirname('')), dirPath)}/${file}`).isDirectory())
```

Comment:
You do all the images in parallel so at some point the GitHub action could run out of memory. Might be better to do it in a serie:

```
for (const file of files) {
 ....
}
```

or chunks

```
for (const files of _.chunk(allFiles, 50)) {
   ...
}
```

but maybe not necessary now and you can fix it once it becomes a problem.